src/vs/workbench/browser/parts/editor/titleControl.ts

```diff
@@ -407,7 +407,7 @@ export abstract class TitleControl extends Themable {
 
 	abstract updateStyles(): void;
 
-	abstract layout(dimension: Dimension): void;
+	abstract layout(dimension: Dimension): number | undefined;
```

Comment:
For consistency I would suggest that `layout` returns both `height` and `width` (as a `Dimension`) and not just the one property. I would think that the `width` property can be used that is passed in because the title widget always gets as wide as what is given.

Reply:
Done, it now returns a `Dimension`. It still has the problem of returning a `Dimension` before the layout call actually happens on the next scheduled frame, but once we figure out how to proceed with that, I'm sure it can be easily fit in.